libs/commons/Textedit.ml

```diff
@@ -53,7 +53,7 @@ let remove_overlapping_edits edits =
   f [] [] edits
 
 let apply_edit_to_text text { start; end_; replacement_text; _ } =
-  let before = Str.first_chars text start in
+  let before = Str.string_before text start in
```

Comment:
Both of these functions achieve the same result. However, I rewrote the function since it is more consistent with the usage of `string_after` on the line below.